### PR TITLE
fix source map with multibyte characters

### PR DESCRIPTION
**Pull request by lwedge99**
## PR description

Make `getCharacterOffsetToLineAndColumnMapping()` handle multibyte characters properly. Previously, it counts all characters as 1 byte.

Addresses https://github.com/trufflesuite/truffle/issues/5914

## Testing instructions

Run `truffle debug 0xfb6027725ccff056d8fe4065227fecd335fe123e8ab4da3a85a71377f7e9c6f9 --fetch-external`. With this bugfix, source mapping seems correct

### Steps
1. Start ganache in terminal 1
  ```console
   ganache --fork
  ```
2. In terminal 2 
  ```console
  truffle debug 0xfb6027725ccff056d8fe4065227fecd335fe123e8ab4da3a85a71377f7e9c6f9 --fetch-external --url http://localhost:8545
  ```
3.  In Truffle debugger
  ```console 
  debug(localhost:8545:0xfb602772...)> b AggregationRouterV5.sol:2849
  debug(localhost:8545:0xfb602772...)>  c
  ```
4. observe source misalignment w/ highlight

### CODE REVIEW

**Comment by gnidan** [comment]
😳 has this been the problem with misaligned source maps the whole time?!

awesome find... thank you for figuring this out!

**Review by gnidan — APPROVED** [review]

File: packages/source-map-utils/index.js
Comment (cds-amal) [inline packages/source-map-utils/index.js:20]: If `loc` is declared and initited, there's no need for the else clause below. 
```javascript
      let loc = { line, column };
```

**Review by cds-amal — APPROVED** [review]
Tested and looks good to me! I'd recommend waiting for @haltman-at to comment before merging.

## error - observed

```console
debug(localhost:8545:0xfb602772...)> b AggregationRouterV5.sol:2849
Breakpoint added at line 2849 in AggregationRouterV5.sol.
debug(localhost:8545:0xfb602772...)> c

AggregationRouterV5.sol:

2847:
2848:     /// @inheritdoc IUniswapV3SwapCallback
2849:     function uniswapV3SwapCallback(
               ^^^^^^^^^^
```

## fix - observed

```console
debug(localhost:8545:0xfb602772...)> b AggregationRouterV5.sol:2849
Breakpoint added at line 2849 in AggregationRouterV5.sol.
debug(localhost:8545:0xfb602772...)> c

AggregationRouterV5.sol:

2847:
2848:     /// @inheritdoc IUniswapV3SwapCallback
2849:     function uniswapV3SwapCallback(
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

debug(localhost:8545:0xfb602772...)>
```

File: packages/source-map-utils/index.js
Comment (haltman-at) [inline packages/source-map-utils/index.js:31]: I'd suggest adding a comment on this loop.  Something like, "because our offsets are given in bytes rather than in characters, we pad out the line/column map as per the UTF-8 length of the character".  I think that will make this clearer for future maintainers.

**Review by haltman-at — COMMENTED** [review]
Thanks for catching this!  My only note is to suggest adding an explanatory comment.  On first read I wasn't convinced this was an appropriate solution.  I had to look it over a bit more to realize that yes this was actually a good way of handling this.  So, I'd like to request adding a comment to explain.

**Comment by haltman-at** [comment]
Actually, I just added the comment myself, so you don't need to do anything.  I'll merge as soon as this passes CI!

**Pull request merged**
